Client/Frontend/Browser/Onboarding/Welcome/WelcomeViewCallout.swift

```diff
@@ -0,0 +1,478 @@
+    case defaultBrowser(title: String, details: String, primaryButtonTitle: String, secondaryButtonTitle: String, primaryAction: () -> Void, secondaryAction: () -> Void)
+    case defaultBrowserWarning(title: String, details: String, primaryButtonTitle: String, secondaryButtonTitle: String, primaryAction: () -> Void, secondaryAction: () -> Void)
+    case ready(title: String, details: String, moreDetails: String)
+}
+
+class WelcomeViewCallout: UIView {
+    private struct DesignUX {
+        static let padding = 20.0
+        static let contentPadding = 30.0
+        static let cornerRadius = 16.0
+    }
+    
+    private let backgroundView = RoundedBackgroundView(cornerRadius: DesignUX.cornerRadius)
+    
+    private let arrowView = CalloutArrowView().then {
+        $0.backgroundColor = .secondaryBraveBackground
+    }
+    
+    private let contentView = UIStackView().then {
+        $0.axis = .vertical
+        $0.layoutMargins = UIEdgeInsets(equalInset: DesignUX.contentPadding)
+        $0.isLayoutMarginsRelativeArrangement = true
+    }
+    
+    // MARK: - Content
+    private let titleLabel = UILabel().then {
+        $0.textColor = .bravePrimary
+        $0.textAlignment = .center
+        $0.numberOfLines = 0
+        $0.setContentHuggingPriority(.required, for: .vertical)
+        $0.setContentCompressionResistancePriority(.required, for: .vertical)
+    }
+    
+    private let detailsLabel = UILabel().then {
+        $0.textColor = .bravePrimary
+        $0.textAlignment = .left
+        $0.numberOfLines = 0
+        $0.setContentHuggingPriority(.required, for: .vertical)
+        $0.setContentCompressionResistancePriority(.required, for: .vertical)
+    }
+    
+    private let primaryButton = RoundInterfaceButton(type: .custom).then {
+        $0.setTitleColor(.white, for: .normal)
+        $0.backgroundColor = .braveBlurple
+        $0.setTitle(" ", for: .normal)
+    }
+    
+    private let secondaryButtonContentView = UIStackView().then {
+        $0.axis = .horizontal
+        $0.spacing = 15.0
+        $0.isHidden = true
+        $0.alpha = 0.0
+        $0.layoutMargins = UIEdgeInsets(top: 0.0, left: 15.0, bottom: 0.0, right: 15.0)
+        $0.isLayoutMarginsRelativeArrangement = true
+    }
+    
+    private let secondaryLabel = UILabel().then {
+        $0.textColor = .bravePrimary
+        $0.textAlignment = .center
+        $0.numberOfLines = 0
+        $0.setContentHuggingPriority(.required, for: .horizontal)
+        $0.setContentCompressionResistancePriority(.required, for: .horizontal)
+        $0.setContentHuggingPriority(.required, for: .vertical)
+        $0.setContentCompressionResistancePriority(.required, for: .vertical)
+    }
+    
+    private let secondaryButton = RoundInterfaceButton(type: .custom).then {
+        $0.setTitleColor(.braveBlurple, for: .normal)
+        $0.backgroundColor = .clear
+        $0.setTitle(" ", for: .normal)
```

Comment:
Maybe add comment above why one space string is needed here

Reply:
Fixed at https://github.com/brave/brave-ios/pull/4508/commits/9f72c586778075cc4dc78b26127c110ec6ce4bb8